Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -469,7 +491,10 @@ static bool validateWriteTexture(const WGPUImageCopyTexture& destination, const
         for (size_t z = 0; z < maxZ; ++z) {
             for (size_t y = 0; y < maxY; ++y) {
                 auto sourceBytes = static_cast<const uint8_t*>(data) + y * bytesPerRow + z * bytesPerImage;
-                RELEASE_ASSERT(y * bytesPerRow + z * bytesPerImage + newBytesPerRow <= dataByteSize);
+                if (y * bytesPerRow + z * bytesPerImage + newBytesPerRow > dataByteSize) {
```

Comment:
Does this need to use `checked` arithmetic or are we certain that `size_t` will never overflow? (eg. Do we build WebGPU on watchOS?)

Reply:
Added the checked arithmetic check outside of the for loop as the maximum values are known prior to entering the loops